### ISSUE TITLE
[Suggestion/Bug] Galaxy Map Fixes

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#3004]
Love the galaxy map, though of course a few things are needed.

1. It would be helpful to be able to use WASD to move from one sector when fully zoomed in to another in addition to clicking

2. Selecting a sector should center the camera on that system. You currently need to zoom all the way in to center your camera

3. Possibly be able to select systems even when zoomed out all the way. It is a bit awkward needing to zoom in a few times to be select, though part of it is because you need to do this to refocus the camera (2, 1.) Maybe you can click and drag when zoomed in to move, but holding the mouse in one place for a short time will enter the system.

4. It would also be helpful to be click on the current system you are in symbol or other marked system and instantly zoom there
![](https://cdn.discordapp.com/attachments/406664114249793536/760761936949674006/unknown.png)

5. It seems that the systems at the very edge are cut in half
![](https://cdn.discordapp.com/attachments/406664114249793536/760752320664961064/unknown.png)

6. You can see the edge of the map on the holoprojector
![](https://cdn.discordapp.com/attachments/406664114249793536/760753247119605780/unknown.png)

7. The squareness of the galaxy is a little weird. Additionally it looks like the top/bottom of the galaxy is cut off. Honestly it reminds me a bit of watching an old movie on a wide-screen tv. I'm not really sure of a good solution the problems are two and fixing one breaks the other or adds many unnecessary systems. You can make it bigger, but that adds additional systems.
![](https://cdn.discordapp.com/attachments/271723356795961345/760752077541998602/unknown.png)

8. The center of the galaxy is way too bright in map
![](https://cdn.discordapp.com/attachments/406664114249793536/760759979111612436/unknown.png)

9. When in 3D mode looking at a system, many stars are too high for the camera to see. They are also above the camera (haha weed joke). Maybe a way to drag-view in system to go up or down or even rotate.
![](https://cdn.discordapp.com/attachments/406664114249793536/760760283584004126/unknown.png)

**Comment by Gliese-832-c** [comment]
As for point 7, the square galaxy, in my opinion, the map should just get bigger. Real life galaxies are huuuge, like, hundreds of thousands of light years huge, no single human mind can imagine such sizes, so reflecting that ingame would be nice.

**Comment by tsunamayo** [comment]
Yeah you pick all the points I still wanted to improve ;) Thanks for the fast feedback. I will repush a new update with the correct patch, my bad. 
@Gliese-832-c I dont want to make a huge galaxy like before, it is just a pain to navigate and code, and serve no purpose really. I leave that to Elite.

**Comment by tsunamayo** [comment]
new build is up!

**Comment by Kaiser-Indrasil** [comment]
I was quick to notice a little problem with the star generation. There are a few stars that get generated along the galaxy's plane which looks a bit obvious:
![image](https://user-images.githubusercontent.com/63756057/94676882-e8354800-0313-11eb-9f07-b89d0005c46b.png)

**Comment by Gliese-832-c** [comment]
@tsunamayo Fair point. I guess it doesn't need to be *huge*. It's just, that to combat the squareness, I would prefer an enlargement of the grid over a shrinking of the existing galaxy.

**Comment by Kaiser-Indrasil** [comment]
I can see you drew a lot of inspiration from Avorion's approach to the galaxy map but their approach to UX is much more convenient for the player.
The sectors over there are just a cosmetic thing and you can move across the galaxy even being fully zoomed in. No need to select them individually.
If you would like to keep the sectors, I would suggest the idea of constellations from Eve Online, where up to 10 stars form a constellation:
![image](https://user-images.githubusercontent.com/63756057/94702947-3b1ff700-0336-11eb-973f-b5f2b11f6f8e.png)
...unless you want everyone's galaxy to be different.

Star generation needs some improvement, too. The amount of stars per sector stays the same either inside or outside a galaxy arm, which looks weird.
![image](https://user-images.githubusercontent.com/63756057/94678280-203d8a80-0316-11eb-9d68-b2b8f6b5508e.png)
Also, nice galaxy omlette :D
![image](https://user-images.githubusercontent.com/63756057/94678654-bd002800-0316-11eb-9686-52d491f4679b.png)
